dialogue-core/src/main/java/com/palantir/dialogue/core/TimingEndpointChannel.java

```diff
@@ -68,20 +70,23 @@ static EndpointChannel create(Config cf, EndpointChannel delegate, Endpoint endp
 
         return DialogueFutures.addDirectCallback(response, new FutureCallback<Response>() {
             @Override
-            public void onSuccess(Response _result) {
-                updateResponseTimer();
+            public void onSuccess(Response response) {
+                if (Responses.isClientSuccess(response)) {
+                    updateTimer(successTimer);
+                } else if (Responses.isQosStatus(response) || Responses.isInternalServerError(response)) {
+                    updateTimer(failureTimer);
+                }
             }
 
             @Override
             public void onFailure(Throwable throwable) {
-                updateResponseTimer();
                 if (throwable instanceof IOException) {
-                    ioExceptionMeter.mark();
+                    updateTimer(failureTimer);
                 }
```

Comment:
Thoughts on adding an else here safe-logging the exception class name at `info` level? That way we can verify we aren't missing non-ioexceptions that should be counted toward failure.

Reply:
are non IOExceptions even allowed here? I will add the logging, do you think if should be rate limited, in case we are actually missing a lot of stuff here?

Reply:
> are non IOExceptions even allowed here?

That's what I'd like to find out :-)

> do you think if should be rate limited, in case we are actually missing a lot of stuff here?

I think logging the exception type rather than the stack trace cuts down enough potential cruft that rate limiting shouldn't be necessary. Any exception thrown here will be logged elsewhere, so +1 log line shouldn't be an issue, but I've been wrong before.

Reply:
I'll add it to log once a second, if it turns up anything after 2 weeks, we can up it or just drive it down.